alf/utils/value_ops.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+
+def discounted_return(
+        rewards,
+        values,
+        is_lasts,
+        discounts,
+        time_major=True):
+    if not time_major:
```

Comment:
If the problem of tf_agents.utils.value_ops.discounted_return is the discount is not 0 when is_last for some environments, we can simply multiply discount with 1-is_last and call the original discounted_return().

Here the is_last is from current time_step. But  is_last should  be from next_step.
And also please comment what's wrong with the original discounted_return() and how discount is specified by tf_agents

Reply:
For TimeLimit Environment , an episode terminate by time exceed or normal end (goal acheived or unable to continue)

1. tfagents set discounts to 1.0 for first case , eg: Pendulum, V(TimeStep; TimeStep.step==StepType.LAST) sould not be zero


2. tfagents set discounts to 0.0 for steps where next_time_step.SteypType==StepType.LAST